snap/validate.go

```diff
@@ -35,3 +35,13 @@ func ValidateName(name string) error {
 	}
 	return nil
 }
+
+// Validate verifies the content in the info.
+func Validate(info *Info) error {
```

Comment:
How do you imagine using those? Should we validate when reading the info in at all times? If so, likely don't need those publicly?